src/caustics/lenses/batchedplane.py

```diff
@@ -0,0 +1,198 @@
+
+    lens: ThinLens
+        A ThinLens object that will be vmapped over into a single lensing plane.
+
+    """
+
+    def __init__(
+        self,
+        cosmology: CosmologyType,
+        lens: ThinLens,
+        name: NameType = None,
+        z_l: ZLType = None,
+    ):
+        """
+        Initialize the SinglePlane lens model.
+        """
+        super().__init__(cosmology, z_l=z_l, name=name)
+        self.lens = lens
+
+    @forward
+    def reduced_deflection_angle(
+        self,
+        x: Tensor,
+        y: Tensor,
+        z_s: Tensor,
+    ) -> tuple[Tensor, Tensor]:
+        """
+        Calculate the total deflection angle by summing
+        the deflection angles of all individual lenses.
+
+        Parameters
+        ----------
+        x: Tensor
+            The x-coordinate of the lens.
+
+            *Unit: arcsec*
+
+        y: Tensor
+            The y-coordinate of the lens.
+
+            *Unit: arcsec*
+
+        z_s: Tensor
+            The source redshift.
+
+            *Unit: unitless*
+
+        params: Packed, optional
+            Dynamic parameter container.
+
+        Returns
+        -------
+        x_component: Tensor
+            The x-component of the deflection angle.
+
+            *Unit: arcsec*
+
+        y_component: Tensor
+            The y-component of the deflection angle.
+
+            *Unit: arcsec*
+
+        """
+
+        # Collect the dynamic parameters to vmap over
+        params = dict((p.name, p.value) for p in self.lens.local_dynamic_params)
+        batchdims = dict(
+            (p.name, -(len(p.shape) + 1)) for p in self.lens.local_dynamic_params
+        )
+        ax, ay = torch.vmap(
+            lambda p: self.lens.reduced_deflection_angle(x, y, z_s, **p),
+            in_dims=(batchdims,),
+        )(params)
+        return ax.sum(dim=0), ay.sum(dim=0)
+
+    @forward
+    def convergence(
+        self,
+        x: Tensor,
+        y: Tensor,
+        z_s: Tensor,
+    ) -> Tensor:
+        """
+        Calculate the total projected mass density by
+        summing the mass densities of all individual lenses.
+
+        Parameters
+        ----------
+        x: Tensor
+            The x-coordinate of the lens.
+
+            *Unit: arcsec*
+
+        y: Tensor
+            The y-coordinate of the lens.
+
+            *Unit: arcsec*
+
+        z_s: Tensor
+            The source redshift.
+
+            *Unit: unitless*
+
+        params: Packed, optional
+            Dynamic parameter container.
+
+        Returns
+        -------
+        Tensor
+            The total projected mass density.
+
+            *Unit: unitless*
+
+        """
+        # Collect the dynamic parameters to vmap over
+        params = dict((p.name, p.value) for p in self.lens.local_dynamic_params)
+        batchdims = dict(
+            (p.name, -(len(p.shape) + 1)) for p in self.lens.local_dynamic_params
+        )
+        convergence = torch.vmap(
```

Comment:
I worry that this operation can consume a lot of memory at its peak. In practice, when drawing convergence, we often deal with large population. If not careful, the GPU memory will be entirely consumed by copies of the plane to draw the halo on. 

In practice, I prefer to use a mix of vmap and for loop to manage the peak memory usage of this operation. This requires defining a batch_size in the init and doing something like
```python
total = params.shape[0]
convergence = 0
for _ in range(total//batch_size):
    convergence += vmap(self.lens.convergence(...)).sum(0)
return convergence
```

Reply:
I think I could just use the `chunk_size` parameter in vmap, but yeah that's a good point I can expose that option to the user